### ISSUE TITLE
[java] UseStandardCharsets: False negative with Java 22+ and UTF-32 charsets

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#6719]
**Affects PMD Version:** 
7.24.0

**Rule:**

<https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#usestandardcharsets>

**Description:**

Java 22 added 32 bit encodings to `StandardCharsets`, these are not supported by the `UseStandardCharsetsRule` rule. The fix for this should depend on `javaLevel`.

**Code Sample demonstrating the issue:**

```java
void main() {
byte[] bytes = "foo".getBytes("UTF-32");
}
```

**Expected outcome:**

PMD should report a violation at line 2, but doesn't. This is a false-negative.

**Running PMD through:** CLI

**Comment by UncleOwen** [comment]
See https://bugs.openjdk.org/browse/JDK-8310047
